### PR TITLE
fix issue #75: add values parameter to all network/interface/*.pp classes

### PR DESCRIPTION
**Pull request by buzzdeee**
The values parameter, an array of strings, is intended to allow the user to set alias IP addresses, as well as shell commands, i.e. setting routes via !commands.

The values array, if given, is concatenated to the returned string of get_hostname_if_<INTERFACE_TYPE> functions, and then handed over to the values parameter of 'bsd::network::interface'

I only need it soon for carp interfaces, but thought all classes in that subdirectory should follow the same pattern, so I added it for all of them.

However, the vlan.pp was not following the pattern of the others, and now the FreeBSD test for vlan interfaces fails. I don't have a FreeBSD system around to test that. Do you have a possibility to verify that it creates a valid working FreeBSD config?
On OpenBSD, the config it created for VLAN interfaces was odd, looking like the following:
```
172.16.0.1/24 vlan 100 vlandev trunk0
 inet 172.16.0.1 255.255.255.0 NONE
 up
```

Now after the change, it creates files like:
```
vlan 100 vlandev trunk0
inet 172.16.0.1 255.255.255.0 NONE
up
```

The odd version it created was working fine, but the version it now creates is from my point of view saner.

In short: do you can let me know if VLANs for FreeBSD are now broken or do they work?
When they work, I'd rather go and adapt the spec test, if I broke it, I'll rework the vlan.pp in 
order to resemble behaviour as it was before.

### CODE REVIEW

**Comment by xaque208** [comment]
I can confirm that VLAN interfaces on FreeBSD do get created correctly.

I'm definitely in favor of this kind of change though there may be a simpler change available.  I've recently updated the VLAN code, which is why it doesn't conform to the broken state of the rest of the interfaces.  

https://github.com/xaque208/puppet-bsd/blob/master/manifests/network/interface/vlan.pp#L38

I think much of my own confusion around this was just wording, so changing the variable to $vlan_options and then passing $vlan_options as the value of the 'options' parameter worked as expected.  Since effectively (I think) the pseudo interface specifics are indeed interface params, options made sense.  This allows the passing of $address straight to the bsd::network::interace define as the 'values' param.

I'm open to feedback here.  I know its not very clear (at least to me) what 'values' is meant to be, but I couldn't really come up with a better word to describe "a bunch of stuff for a bunch of scenarios'.

Anyway, keeping $address as the interface to set the values on the sub interface types helps at least readability of the code I think.

Thanks for digging in, I appreciate your efforts.

**Comment by buzzdeee** [comment]
While trying to 'fix' the FreeBSD vlan spec test, I figured, that the 'address' parameter is not propagated into what is generated in rc.conf or whereever it ends up in FreeBSD.
Since the 'address' is optional, it doesn't blow up.

I have to dig a bit deeper, and I think I already have an idea, just not yet had time to test if it works out ;). 
Anyways, for that, it would help, if you could give an example, of what the result of a FreeBSD vlan device configuration with an IP address would look like.

thanks,
Sebastian

**Comment by buzzdeee** [comment]
The idea I had in mind, didn't worked out. The bsd::network::interface defined type should have an address parameter, instead of giving the addresses in the values.
Something I'll look into later on. 
Therefore I kept the VLAN defined type a bit differently, to make the spec tests happy. 

For whatever reason, there now seems to be a conflict in the vlan.pp :(

Sebastian

**Comment by xaque208** [comment]
It looks like you've worked out the bit for FreeBSD vlans, just casing around it.  I'm fine with that for now.  Thank you for the efforts!
